overlays/nodejs/serverless/function.ts

```diff
@@ -309,18 +313,20 @@ function addPackageAndDependenciesToSet(s: Set<string>, root: Package, pkg: stri
 // It is assumed that the tree was correctly construted such that dependencies are resolved to compatible versions in
 // the closest available match starting at the provided root and walking up to the head of the tree.
 function findDependency(root: Package, name: string) {
-    for(;root;root = root.parent) {
+    for(; root; root = root.parent) {
         for (var child of root.children) {
-            let childName = child.name;
-            // Note: `read-package-tree` returns incorrect `.name` properties for packages in an orgnaization - like
-            // `@types/express` or `@protobufjs/path`.  Compute the correct name from the `path` property instead.
-            // Match any name that ends with something that looks like `@foo/bar`.
-            const match = /\/\@[^\/]*\/[^\/]*$/.exec(child.path);
-            if (match) {
-                childName = match[0];
-            }
-            if(childName === name) {
-                return child;
+            if (name.indexOf("@") === -1) {
```

Comment:
Perhaps I'm misunderstanding, but it seems to do the right thing.

name is the package being sought (because it was used in a `require` or `import`).  Note that, importantly, this is *not* the `child.name`, which does not have a `@` in it.  During an import of an org-scoped package, `name` begins with a `@`.  It's only then that we need to fall into the logic to match the org and name separately.

For example, if we're looking for `@slack/client`, then `name` will be `"@slack/client"` and `child.path` will be something like `node_modules/@slack/client`.  In that case, we split the `child.path` using the filesystem separator and concatenate the last two parts, to produce `"@slack/client"`.

If we're looking for a simple name, like `semver`, we just compare the names directly.

Reply:
Ahh - yes - I was thinking this was testing `child.name`.